src/main/java/com/monumental/util/csvparsing/CsvMonumentConverter.java

```diff
@@ -18,159 +20,157 @@
+                        } catch (Exception e) {
+                            result.getWarnings().add("Date should be a valid date in the format DD-MM-YYYY or YYYY.");
+                        }
+                        break;
+                    case "materials":
+                        result.getMaterialNames().addAll(parseTags(value));
+                        break;
+                    case "inscription":
+                        monument.setInscription(value);
+                        break;
+                    case "latitude":
+                        try {
+                            latitude = Double.parseDouble(value);
+                        } catch (NumberFormatException e) {
+                            result.getWarnings().add("Latitude should be a valid number.");
+                        }
+                        break;
+                    case "longitude":
+                        try {
+                            longitude = Double.parseDouble(value);
+                        } catch (NumberFormatException e) {
+                            result.getWarnings().add("Longitude should be a valid number.");
+                        }
+                        break;
+                    case "city":
+                        monument.setCity(value);
+                        break;
+                    case "state":
+                        monument.setState(value);
+                        break;
+                    case "address":
+                        monument.setAddress(value);
+                        break;
+                    case "tags":
+                        result.getTagNames().addAll(parseTags(value));
+                        break;
+                    case "reference":
+                        Reference reference = parseReference(value);
+                        reference.setMonument(monument);
+                        monument.getReferences().add(reference);
+                        break;
+                    case "images":
+                        if (zipFile != null) {
+                            ZipEntry imageZipEntry = zipFile.getEntry(value);
+                            if (imageZipEntry == null) {
+                                result.getWarnings().add("Could not find image in .zip file. File may be missing or named incorrectly.");
+                            } else {
+                                try {
+                                    result.getImageFiles().add(ZipFileHelper.convertZipEntryToFile(zipFile, imageZipEntry));
+                                } catch (IOException e) {
+                                    result.getErrors().add("Failed to read image file from .zip");
+                                }
+                            }
+                        } else {
+                            result.getWarnings().add("Cannot upload images with a .csv file. You must package your .csv and your images into a .zip file and upload it.");
```

Comment:
Maybe we should just add this at the top of the page instead of displaying it as a warning for each row.

Reply:
V2